openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -136,11 +140,26 @@ public void executeTool(ToolContext toolContext) {
             }
         } catch (BallerinaOpenApiException e) {
             Constants.DiagnosticMessages error = Constants.DiagnosticMessages.PARSER_ERROR;
-            reportDiagnostics(error, e.getMessage(), location);
+            createDiagnostics(toolContext, error, e.getMessage(), location);
         } catch (IOException | FormatterException e) {
             Constants.DiagnosticMessages error = Constants.DiagnosticMessages.ERROR_WHILE_GENERATING_CLIENT;
-            reportDiagnostics(error, e.getMessage(), location);
+            createDiagnostics(toolContext, error, e.getMessage(), location);
+        }
+    }
+
+    /**
+     * This method uses to validate the cache.
+     */
+    private static boolean validateCache(ToolContext toolContext, OASClientConfig clientConfig) throws IOException {
```

Comment:
do we need this to be static?

Reply:
Made it static method because it doesn't rely on instance-specific data.

Reply:
Making these static has made the `md5HexOpenAPI` variable also static. When there are multiple openapi contracts defined in Ballerina.toml, we call this `OpenAPICodeGeneratorTool` multiple times right? In that case, executions of `OpenAPICodeGeneratorTool` will see cache values of other contracts as well right? I think there is no possibility for any misuse of that value. Hence this is also ok i think.